tests/components/bbox/test_device_tracker.py

```diff
@@ -0,0 +1,177 @@
+
+
+async def test_get_device_name(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test getting device name by MAC address."""
+    # Create scanner without calling __init__ to avoid API calls
+    scanner = BboxDeviceScanner.__new__(BboxDeviceScanner)
+    scanner.host = device_tracker_config[DEVICE_TRACKER_DOMAIN]["host"]
+    scanner.success_init = True
+    scanner.last_results = [
+        Device("aa:bb:cc:dd:ee:ff", "test_device", "192.168.1.100", None),
+        Device("ff:ee:dd:cc:bb:aa", "another_device", "192.168.1.101", None),
+    ]
+
+    # Test existing device
+    name = scanner.get_device_name("aa:bb:cc:dd:ee:ff")
+    assert name == "test_device"
+
+    # Test non-existing device
+    name = scanner.get_device_name("11:22:33:44:55:66")
+    assert name is None
+
+
+async def test_update_info_success(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test successful device info update."""
+    with patch("homeassistant.components.bbox.device_tracker.pybbox") as mock_pybbox:
+        mock_box = MagicMock()
+        mock_pybbox.Bbox.return_value = mock_box
+        mock_box.get_all_connected_devices.return_value = [
+            {
+                "macaddress": "aa:bb:cc:dd:ee:ff",
+                "hostname": "test_device",
+                "ipaddress": "192.168.1.100",
+                "active": 1,
+            },
+            {
+                "macaddress": "ff:ee:dd:cc:bb:aa",
+                "hostname": "inactive_device",
+                "ipaddress": "192.168.1.101",
+                "active": 0,  # Should be filtered out
+            },
+        ]
+
+        # Create scanner without calling __init__ to avoid API calls
+        scanner = BboxDeviceScanner.__new__(BboxDeviceScanner)
+        scanner.host = device_tracker_config[DEVICE_TRACKER_DOMAIN]["host"]
+        scanner.success_init = True
+        scanner.last_results = []
+
+        scanner._update_info()
+
+        # _update_info doesn't return a value on success, just updates last_results
+        assert len(scanner.last_results) == 1
+        device = scanner.last_results[0]
+        assert device.mac == "aa:bb:cc:dd:ee:ff"
+        assert device.name == "test_device"
+        assert device.ip == "192.168.1.100"
+
+
+async def test_update_info_failure(
+    hass: HomeAssistant, device_tracker_config: ConfigType
+) -> None:
+    """Test device info update failure."""
+    with patch("homeassistant.components.bbox.device_tracker.pybbox") as mock_pybbox:
+        mock_box = MagicMock()
+        mock_pybbox.Bbox.return_value = mock_box
+        mock_box.get_all_connected_devices.side_effect = requests.exceptions.HTTPError(
+            "API Error"
+        )
+
+        # Create scanner without calling __init__ to avoid API calls
+        scanner = BboxDeviceScanner.__new__(BboxDeviceScanner)
+        scanner.host = device_tracker_config[DEVICE_TRACKER_DOMAIN]["host"]
+        scanner.success_init = True
```

Comment:
It looks like private property. We should setting value on private property as it makes refactor harder.